### PR TITLE
feat: add support for handlebars templating engine as an option

### PR DESCRIPTION
**Pull request by erise133**
# Relates to:
Added support for the Handlebars templating engine in the composeContext function, allowing for advanced template processing in addition to the existing simple string replacement.

# Risks
- Existing functionality for simple string replacement remains unchanged.
- Handlebars is an optional addition and is only used when specified, minimizing potential impact on existing use cases.

# Background

## What does this PR do?
This PR enhances the composeContext function by integrating Handlebars as an optional templating engine. Users can now choose between simple key replacement or advanced template features provided by Handlebars, such as conditionals and loops.

## What kind of change is this?
- Feature: Adds a new functionality to support Handlebars templating engine.

# Documentation changes needed?
- Yes: Update documentation to include examples of using the Handlebars templating engine with composeContext.

# Testing

## Where should a reviewer start?
- Review the changes in context.ts and the newly added tests in context.test.ts.

## Detailed testing steps
- Run the tests in context.test.ts to verify both the simple string replacement and Handlebars functionality.
- Ensure all scenarios, including edge cases like missing values and advanced Handlebars templates, are covered by tests.

### CODE REVIEW

File: packages/core/src/context.ts [hunk 1]
```diff
@@ -7,27 +8,70 @@ import { type State } from "./types.ts";
...
  * @example
  * // Given a state object and a template
  * const state = { userName: "Alice", userAge: 30 };
  * const template = "Hello, {{userName}}! You are {{userAge}} years old";
  *
- * // Composing the context will result in:
+ * // Composing the context with simple string replacement will result in:
  * // "Hello, Alice! You are 30 years old."
- * const context = composeContext({ state, template });
+ * const contextSimple = composeContext({ state, template });
```
Comment (monilpat) [inline packages/core/src/context.ts:26]: Can this comment be moved to a Readme instead thanks :)

File: packages/core/src/tests/context.test.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import { describe, expect, it } from "vitest";
+import { composeContext } from "../context";
+import handlebars from "handlebars";
+import { State } from "../types.ts";
+
+describe("composeContext", () => {
```
Comment (monilpat) [inline packages/core/src/tests/context.test.ts:6]: Thanks for the test :)

**Review by monilpat — APPROVED** [review]
LGTM thanks for doing this :)

**Review by chrislatorres — APPROVED** [review]
LGTM

I agree with @monilpat on the long comment being moved somewhere else.

**Comment by chrislatorres** [comment]
@erise133 

 you should be able to get a coder role from the #⁠roles channel in Discord.
